Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -1725,12 +1725,18 @@ extension SourceKitServer {
     }
 
     let resolved = self.extractIndexedOccurrences(symbols: symbols, index: index, useLocalFallback: true) {
-      (usr, index) in
+      (symbolDetails, index) in
+      guard let usr = symbolDetails.usr else { return [] }
       logger.info("performing indexed jump-to-def with usr \(usr)")
       var occurs = index.occurrences(ofUSR: usr, roles: [.definition])
       if occurs.isEmpty {
         occurs = index.occurrences(ofUSR: usr, roles: [.declaration])
       }
+      if symbolDetails.isDynamic {
+        occurs += occurs.flatMap {
+          index.occurrences(relatedToUSR: $0.symbol.usr, roles: .overrideOf)
```

Comment:
Ideally we should filter by receivers here, ie.
  - Get the set of types to filter (receivers + their subtypes)
  - For each override check if it's a child of one of the types in the filter

Reply:
Added a filter by receiver types.